chia/server/server.py

```diff
@@ -265,12 +281,20 @@ async def garbage_collect_connections_task(self) -> None:
             for peer_ip in to_remove_ban:
                 del self.banned_peers[peer_ip]
 
+    def add_peer(self, peer: PeerInfo) -> None:
+        if self._reconnect_tasks.get(peer) is not None:
+            raise ServerError(f"Peer {peer} already added")
+        self._reconnect_tasks[peer] = asyncio.create_task(self._reconnect_task_handler(peer))
```

Comment:
do you know why we have a reconnect task for each peer? It seems really wasteful for every peer to iterate over all peers just to find itself. It would seem simpler an more efficient to just have one task that iterates over all peers and retries the ones that should be retried